pkg/sessionctx/context.go

```diff
@@ -124,6 +124,9 @@ type Context interface {
 
 	GetSessionManager() util.SessionManager
 
+	// GetAuthPlugins returns all the registered authentication plugins
+	GetAuthPlugins() map[string]*extension.AuthPlugin
```

Comment:
You can move `GetAuthPlugins` to `extension.SessionExtensions` which can be get from `GetExtensions` method of `sessionctx.Context`.

Another question is why we need a `GetAuthPlugins` in session without just call `extension.GetExtensions().GetAuthPlugins()` ?

Reply:
`extension.GetExtensions()` requires locking which could add perf overhead when executing it for every request. Executing it once at connection establishment time is fine.

I will try to move `GetAuthPlugins` into `extension.SessionExtensions`